Adding Section on Terminals (#1186)
ihmeuw/vivarium_research: docs/source/onboarding_resources/computing/index.rst

the same tasks. 

Some of the common languages are Linux, Mac, and Windows. Linux and Mac languages are both derived 
from a parent language Unix and so are very similar. Linux/Mac is most common language at IHME 
and the command line trainings are Linux-based. For this reason, we recommend using a 
terminal that works in Linux or Mac depending on your computer type.

**Recommended Terminals for Windows Users:** 

For updating Vivarium docs and interacting with GitHub, we recommend using Git Bash. This 
installs automatically when you download Git. However, some prefer 
the Windows Subsystem for Linux (WSL) for it's user interface and tools. 

Link to download `Git and Git Bash <https://git-scm.com/>`_

Link to download `WSL <https://learn.microsoft.com/en-us/windows/wsl/install>`_

For accessing the cluster, we recommend using PuTTy or Bitvise. How to access the cluster 
is included in more depth in Section 3. 

**Recommended Terminals for Mac Users:** 

For updating Vivarium docs and interacting with GitHub, we recommend using the Terminal app 
which comes pre-installed on your computer. 

> For accessing the cluster, we recommend using iTerm2. How to access the cluster 

Author: Nathaniel would love your thoughts here but I have updated this to recommend Terminal for everything and have iTerm2 as an additional option.

> For accessing the cluster, we recommend using iTerm2. How to access the cluster 

Reviewer: That sounds good. Yes, Terminal works just fine, but iTerm2 is a nice option if you want more features.

2.0 How to Access a Terminal
++++++++++++++++++++++++++++

A lot of work you will do requires you to access a terminal. A terminal is a way for you, a 
user, to communicate with a computer or computing system. There are many options for 
terminals. This section is written for someone not familiar with terminals, so if 
you have familiarity or favorite software please feel free to use those! 

Writing code into a terminal is called using the command line. IHME has a helpful training 
on the command line in Module 1 of the IHME Learn training `Computational Infrastructure Level 1 <https://ihme.brightspace.com/d2l/home/7028>`_. Command lines technically interact with an 
operating system on your computer. 

> Operating systems are things like Windows or Mac that help a computer run and provide some basic 

Reviewer: ```suggestion
Operating systems are things like Windows or macOS that help a computer run and provide some basic 
```
Mac is the computer, the operating system that runs it is macOS.

Author: Thank you!

infrastructure. Opening and saving files all happen through your operating system. You can change or 
install multiple operating systems, but that is quite uncommon. 

Some operating systems require different command lines to accomplish the same task. The IHME 
cluster uses the operating system Linux, and so the trainings provided are designed for Linux. 

> Linux and Mac are very similar, but Windows is quite different. Therefore, we recommend for 

Reviewer: ```suggestion
Linux and macOS are very similar (they are both `Unix-like <https://en.wikipedia.org/wiki/Unix-like>`_), but Windows is quite different. Therefore, we recommend for 
```

Windows users to install a terminal that can accept command lines written for a Linux while 
still allowing you to still "speak" with your Windows computer. 

**Recommended Terminals for Windows Users:** 

For updating Vivarium docs and interacting with GitHub, we recommend using Git Bash. This 
installs automatically with Git for Windows. However, some prefer 
the Windows Subsystem for Linux (WSL) for its user interface and tools. 

Link to download `Git and Git Bash <https://git-scm.com/>`_

Link to download `WSL <https://learn.microsoft.com/en-us/windows/wsl/install>`_

For accessing the cluster, we recommend using PuTTy or Bitvise. How to access the cluster 
is included in more depth in Section 3. 

**Recommended Terminals for Mac Users:** 

> Since Mac is similar to Linux and is the base operating system on Mac computers, the 

Reviewer: ```suggestion
Since macOS is similar to Linux and is the base operating system on Mac computers, the 
```

pre-installed Terminal app can be used for all your terminal needs. 

**Other Options:**

Most terminals can also be used to access the cluster, although the common practice on 

> the Sim Science team is to use separate terminals for GitHub on our local machine and 

Reviewer 2: Should "GitHub" here be "Git Bash"?
```suggestion
the Sim Science team is to use separate terminals for Git Bash on our local machine and 
```

Reviewer 3: I think the meaning intended here is "to work with GitHub on our local machine," while Git Bash _is_ the terminal used to do this.

```suggestion
the Sim Science team is to use separate terminals for working on our local machine and 
```

Author: Yes - thanks I did clarify this to be working on your local machine.

> the Sim Science team is to use separate terminals for GitHub on our local machine and 

Reviewer: Should we specify that this is the common practice of _Windows_ users on our team? On my Mac, I use the same terminal program for working locally and for connecting to the cluster. I think this is why I've always been confused about all you Windows users talking about Git Bash vs. PuTTY vs. Bitvise...

What about you, @zmbc, do you use multiple terminal programs or just one?

Author: Thanks, I clarified that this is for Windows users only. That also matches with the recommendations we are now giving above.

PuTTY and Bitvise are both SSH clients with graphical user interfaces for Windows. For Mac, it looks like some exist (e.g. Termius) but I didn't see anything super popular.

> We should note that you can also just run ssh commands from a terminal without installing anything instead of using a graphical client. I do this, and I am guessing that Nathaniel does too. Include the link to the IHME training video on how to do this. 

Reviewer 2: Maybe there is an edit to this line that will make it easier for future readers to understand.  Is Syl the speaker referred to as "I" in this case?

Reviewer 3: I think I am the "I" here 😄

Reviewer: Yes, I just `ssh` in from the terminal (technically I [mosh](https://mosh.org/) in), without using a GUI.

Author: Haha sorry - this should probably be in a "note" but some sections of this doc are mainly in note format. I am tackling one section at a time to go over, add, and clarify information. Cluster will be my next section so stay tuned for the next PR! 

And yes - Zeb is the "I" here! It is a copy paste of a comment he left so I didn't lose track of it.
